client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -43,282 +38,269 @@ export async function drag(testId: any, start: any, end: any, lasso = false) {
   await page.mouse.up();
 }
 
-export async function scroll({testId, deltaY, coords}: {testId: string, deltaY: number, coords: number[]}) {
-  const layout = await waitByID(testId);
-  if (layout){
-    const x = coords[0];
-    const y = coords[1];
-    await page.mouse.move(x, y);
-    await page.mouse.down();
-    await page.mouse.up();
-    await page.mouse.wheel({ deltaY });
-  }
+export async function scroll({
+  testId,
+  deltaY,
+  coords,
+  page,
+}: {
+  testId: string;
+  deltaY: number;
+  coords: number[];
+  page: Page;
+}): Promise<void> {
+  await page.getByTestId(testId).waitFor();
+  const x = coords[0];
+  const y = coords[1];
+  await page.mouse.move(x, y);
+  await page.mouse.down();
+  await page.mouse.up();
+  await page.mouse.wheel(0, deltaY);
 }
 
-export async function keyboardUndo() {
-  await page.keyboard.down("MetaLeft");
-  await page.keyboard.press("KeyZ");
-  await page.keyboard.up("MetaLeft");
+export async function keyboardUndo(page: Page): Promise<void> {
+  await page.getByTestId("layout-graph").press("Meta+Z");
 }
 
-export async function keyboardRedo() {
-  await page.keyboard.down("MetaLeft");
-  await page.keyboard.down("ShiftLeft");
-  await page.keyboard.press("KeyZ");
-  await page.keyboard.up("ShiftLeft");
-  await page.keyboard.up("MetaLeft");
+export async function keyboardRedo(page: Page): Promise<void> {
+  await page.getByTestId("layout-graph").press("Meta+Shift+Z");
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function clickOnCoordinate(testId: any, coord: any) {
-  const layout = await expect(page).toMatchElement(getTestId(testId));
-  const elBox = await layout.boxModel();
+const BLUEPRINT_SKELETON_CLASS_NAME = Classes.SKELETON;
+
+export async function waitUntilNoSkeletonDetected(page: Page): Promise<void> {
+  await tryUntil(
+    async () => {
+      const skeleton = await page
+        .locator(`.${BLUEPRINT_SKELETON_CLASS_NAME}`)
+        .all();
+      expect(skeleton).toHaveLength(0);
+    },
+    { page, timeoutMs: 10000 }
```

Comment:
I think we can do this now lol?
```suggestion
    { page, timeoutMs: 10_000 }
```